components/ModelLicense/licenseUtils.ts

```diff
@@ -0,0 +1,175 @@
+import type { CharacterModelLicenseSerializer } from '@/types/Response';
+import type { VRM1Meta } from '@pixiv/three-vrm';
+
+type LicenseProperty = keyof CharacterModelLicenseSerializer;
+
+/**
+ * VRoid Hubで設定出来るVRM0のライセンスから表示テキストを返す
+ * @param license
+ * @param property
+ * @returns
+ */
+export const getHubVrm0LicenseValue = (license: CharacterModelLicenseSerializer, property: LicenseProperty): string => {
+  const status = license[property];
+
+  if (status === 'default') return '未設定';
+
+  switch (property) {
+    case 'characterization_allowed_user':
+      return status === 'everyone' ? 'OK' : 'NG';
+    case 'personal_commercial_use':
+      if (status === 'profit') return 'OK';
+      if (status === 'nonprofit') return '非営利のみ';
+      return 'NG';
+    case 'credit':
+      return status === 'necessary' ? '必要' : '不要';
+    case 'modification':
+      return status === 'allow' ? 'OK' : 'NG';
+    case 'redistribution':
+      return status === 'allow' ? 'OK' : 'NG';
+    case 'sexual_expression':
+      return status === 'allow' ? 'OK' : 'NG';
+    case 'violent_expression':
+      return status === 'allow' ? 'OK' : 'NG';
+    case 'corporate_commercial_use':
+      return status === 'allow' ? 'OK' : 'NG';
+    default:
+      return status === 'allow' ? 'OK' : 'NG';
+  }
+};
+
+/**
+ * VRoid Hubで設定出来るVRM0のライセンスからテキストの表示色の種類を返す
+ * @param license
+ * @param property
+ * @returns
+ */
+export const getHubVrm0LicenseVariant = (
+  license: CharacterModelLicenseSerializer,
+  property: LicenseProperty,
+): 'default' | 'allow' | 'bold' => {
+  const status = license[property];
+
+  switch (property) {
+    case 'characterization_allowed_user':
+      return status === 'everyone' ? 'allow' : 'default';
+    case 'personal_commercial_use':
+      return status === 'disallow' || status === 'default' ? 'default' : 'allow';
```

Comment:
ここだけ`'default'`がデフォルト扱いになっていない点が気になりました。
他の項目ように必要な要素のみ`'allow'`にするのであればのようになるかと思います。

 
```suggestion
      return status === 'profit' || status === 'nonprofit' ? 'allow' : 'default'
```

Reply:
https://github.com/pixiv/VRoidHub-API-Example/pull/7/commits/2e31e828648c23ec6be4e06a4816b528bed9b64b にて修正しました